abi/src/main/java/org/web3j/abi/TypeReference.java

```diff
@@ -88,4 +122,81 @@ public int getSize() {
             return size;
         }
     }
+
+    public static TypeReference makeTypeReference(String solidityType) throws
+            ClassNotFoundException {
+        return makeTypeReference(solidityType,false);
+    }
+
+    public static TypeReference makeTypeReference(String solidityType, final boolean indexed)
+            throws ClassNotFoundException {
+        Matcher m = ARRAY_SUFFIX.matcher(solidityType);
+        if (!m.find()) {
+            final Class tc = getAtomicTypeClass(solidityType);
+            return create(tc, indexed);
+        }
+        String arraySize = m.group(1);
+        TypeReference baseTr = makeTypeReference(solidityType.substring(0,solidityType.length()
+                - m.group(0).length()));
+        TypeReference<?> ref;
+        if (arraySize == null || arraySize.equals("")) {
+            ref = new TypeReference<DynamicArray>(indexed) {
+                @Override
+                public java.lang.reflect.Type getType() {
+                    return new ParameterizedType() {
+                        @Override
+                        public java.lang.reflect.Type[] getActualTypeArguments() {
+                            return new java.lang.reflect.Type[]{baseTr.getType()};
+                        }
+
+                        @Override
+                        public java.lang.reflect.Type getRawType() {
+                            return DynamicArray.class;
+                        }
+
+                        @Override
+                        public java.lang.reflect.Type getOwnerType() {
+                            return Class.class;
+                        }
+                    };
+                }
+            };
+        } else {
+            final Class arrayclass;
+            if (Integer.parseInt((arraySize)) <= StaticArray.MAX_SIZE_OF_STATIC_ARRAY) {
+                arrayclass = Class.forName("org.web3j.abi.datatypes.generated.StaticArray"
+                        + arraySize);
+            } else {
+                arrayclass = Class.forName("org.web3j.abi.datatypes.StaticArray");
```

Comment:
Should not an error be reported instead of using and abstract type?

Reply:
ha that's what I think, but there's an existing unit test that was failing because uint[33] should return StaticArray! That was the existing behavior. I can remove the existing unit test.